diesel_tests/tests/having.rs

```diff
@@ -0,0 +1,154 @@
+            "INSERT INTO comments (id, post_id, text) VALUES (1, 1, 'Comment for Hi Sean'), \
+        (2, 2, 'Comment for Hi Tess'), (3, 2, 'Another comment for Hi Tess')",
+        )
+        .unwrap();
+
+    let source = users::table
+        .inner_join(posts::table.inner_join(comments::table))
+        .group_by((users::id, posts::id))
+        .having(diesel::dsl::count(comments::id).eq(2))
+        .select((users::name, posts::title));
+
+    let expected_data = vec![("Tess".to_string(), "Hi Tess".to_string())];
+    let data: Vec<(String, String)> = source.load(&connection).unwrap();
+
+    assert_eq!(expected_data, data);
+}
+
+#[test]
+fn boxed_simple_having_with_group_by() {
+    let connection = connection();
+    connection
+        .execute("INSERT INTO users (id, name) VALUES (1, 'Sean'), (2, 'Tess')")
+        .unwrap();
+    connection
+        .execute(
+            "INSERT INTO posts (id, user_id, title) VALUES (1, 1, 'Hi Sean'), (2, 2, 'Hi Tess')",
+        )
+        .unwrap();
+    connection
+        .execute(
+            "INSERT INTO comments (id, post_id, text) VALUES (1, 1, 'Comment for Hi Sean'), \
+        (2, 2, 'Comment for Hi Tess'), (3, 2, 'Another comment for Hi Tess')",
+        )
+        .unwrap();
+
+    let source = users::table
+        .inner_join(posts::table.inner_join(comments::table))
+        .group_by((users::id, posts::id))
+        .having(diesel::dsl::count(comments::id).eq(2))
+        .select((users::name, posts::title))
+        .into_boxed();
+
+    let expected_data = vec![("Tess".to_string(), "Hi Tess".to_string())];
+    let data: Vec<(String, String)> = source.load(&connection).unwrap();
+
+    assert_eq!(expected_data, data);
+}
+
+#[test]
+fn multi_condition_having_with_group_by() {
+    let connection = connection();
+    connection
+        .execute("INSERT INTO users (id, name) VALUES (1, 'Sean'), (2, 'Tess'), (3, 'Nick')")
+        .unwrap();
+    connection
+        .execute(
+            "INSERT INTO posts (id, user_id, title) VALUES (1, 1, 'Hi Sean'), (2, 2, 'Hi Tess'), (3, 3, 'Hi Nick')",
+        )
+        .unwrap();
+    connection
+        .execute(
+            "INSERT INTO comments (id, post_id, text) VALUES (1, 1, 'Comment for Hi Sean'), \
+        (2, 2, 'Comment for Hi Tess'), (3, 2, 'Another comment for Hi Tess'), \
+        (4, 3, 'Comment for Hi Nick'), (5, 3, 'Another comment for Hi Nick')",
+        )
+        .unwrap();
+
+    let source = users::table
+        .inner_join(posts::table.inner_join(comments::table))
+        .group_by((users::id, posts::id))
+        .having(diesel::dsl::count(comments::id).eq(2).and(users::id.eq(3)))
+        .select((users::id, users::name, posts::title));
+
+    let expected_data = vec![(3, "Nick".to_string(), "Hi Nick".to_string())];
+    let data: Vec<(i32, String, String)> = source.load(&connection).unwrap();
+
+    assert_eq!(expected_data, data);
+}
+
+#[test]
+fn boxed_multi_condition_having_with_group_by() {
```

Comment:
@weiznich Is this test not a valid test for boxed?

Reply:
This test first sets the having clause and the boxes the query. I'm looking for another test that calls `.into_boxed()` before calling `.having()`, so that we call into the `HavingDsl` of `BoxedSelectStatement`. (Note that you need to call `.into_boxed()` after calling `.group_by`, because changing the grouping is not supported on boxed queries)